### ISSUE TITLE
Problem when installing taxcalc package

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/Tax-Calculator#1222]
I currently have on my Mac computer only the Tax-Calculator source code.  As an experiment, I tried to install the latest taxcalc package using the following command:
```
conda install -c ospc taxcalc
```
Here is what happened next:
```
Fetching package metadata ...........
Solving package specifications: .

Package plan for installation in environment /Users/mrh/anaconda:

The following NEW packages will be INSTALLED:

    taxcalc:      0.7.82-np110py27_0 ospc

The following packages will be UPDATED:

    conda:        4.3.13-py27_0           --> 4.3.14-py27_0     
    llvmlite:     0.7.0-py27_3            --> 0.11.0-py27_0     
    numba:        0.21.0-np19py27_0       --> 0.26.0-np110py27_0

The following packages will be DOWNGRADED due to dependency conflicts:

    numexpr:      2.6.1-np111py27_1       --> 2.6.0-np110py27_0 
    numpy:        1.11.2-py27_0           --> 1.10.4-py27_2     
    scikit-learn: 0.18.1-np111py27_0      --> 0.17.1-np110py27_2
    scipy:        0.18.1-np111py27_0      --> 0.17.1-np110py27_1

Proceed ([y]/n)? n

Exiting
```
I aborted the install because it was going to **downgrade** my version of numpy.

I have on my computer numpy version 1.11.2, which is the version called for in the `tax-calculator/environment.yml` file.

Why is the `conda install -c ospc taxcalc` command trying to **downgrade** my version of numpy?

This seems like a bug.  If it is not a bug, then I need to learn more about how all this conda magic works, which I am eager to do.

@MattHJensen @PeterDSteinberg @zrisher

**Comment by PeterDSteinberg** [comment]
@martinholmer - We had discussed before that there was an inconsistency between conda.recipe and environment.yml and I did not follow to fix that inconsistency.  The PR #1223 is a fix.

**Comment by martinholmer** [comment]
@PeterDSteinberg, Thanks for fixing #1222 in #1223. 
When will the new corrected taxcalc package be available?

@MattHJensen

**Comment by MattHJensen** [comment]
@martinholmer, we'll be able to release the new taxcalc package after I tag the latest version (0.7.9) on GitHub. I think it would make sense to do so as soon as #1221 is merged.

**Comment by martinholmer** [comment]
@MattHJensen, I just merged #1221, so you can make another release of the taxcalc package.

**Comment by MattHJensen** [comment]
@martinholmer, taxcalc 0.7.9 is now available https://anaconda.org/ospc/taxcalc.

**Comment by martinholmer** [comment]
@MattHJensen said:
> taxcalc 0.7.9 is now available at https://anaconda.org/ospc/taxcalc

But isn't this release broken because it doesn't include the bug fix in #1229?

**Comment by MattHJensen** [comment]
I meant for 0.7.9 to include #1229. I can check in a couple of hours.

On Mar 8, 2017, at 9:55 AM, Martin Holmer <notifications@github.com<mailto:notifications@github.com>> wrote:


@MattHJensen<https://github.com/MattHJensen> said:

taxcalc 0.7.9 is now available at https://anaconda.org/ospc/taxcalc

But isn't this release broken because it doesn't include the bug fix in #1229<https://github.com/open-source-economics/Tax-Calculator/pull/1229>?

**Comment by MattHJensen** [comment]
Ahh, I see the problem now. I'll tag 0.7.91 this afternoon -- or also @martinholmer feel free to do it sooner.

On Mar 8, 2017, at 9:55 AM, Martin Holmer <notifications@github.com<mailto:notifications@github.com>> wrote:


@MattHJensen<https://github.com/MattHJensen> said:

taxcalc 0.7.9 is now available at https://anaconda.org/ospc/taxcalc

But isn't this release broken because it doesn't include the bug fix in #1229<https://github.com/open-source-economics/Tax-Calculator/pull/1229>?

**Comment by PeterDSteinberg** [comment]
@MattHJensen - see my comment in flowdock related to tagging minor versions like `0.7.91` - I think we should delete the tag `0.7.9` instead.

**Comment by MattHJensen** [comment]
@PeterDSteinberg I just deleted and retagged `0.7.9`

**Comment by MattHJensen** [comment]
@martinholmer, 0.7.9 includes #1229 now. https://anaconda.org/ospc/taxcalc

**Comment by martinholmer** [comment]
@MattHJensen said:
> 0.7.9 includes #1229 now. https://anaconda.org/ospc/taxcalc

Thanks.

**Comment by martinholmer** [comment]
@MattHJensen said in issue #1222:
> [taxcalc package] 0.7.9 includes #1229 now.   https://anaconda.org/ospc/taxcalc

I'm still having problems.  Here is what happened when I tried to install:
```
iMac2:tc mrh$ conda install -c ospc taxcalc=0.7.9
Fetching package metadata ...........
Solving package specifications: .

Package plan for installation in environment /Users/mrh/anaconda:

The following NEW packages will be INSTALLED:

    taxcalc:      0.7.9-np110py27_0  ospc

The following packages will be UPDATED:

    conda:        4.3.13-py27_0           --> 4.3.14-py27_0     
    llvmlite:     0.7.0-py27_3            --> 0.11.0-py27_0     
    numba:        0.21.0-np19py27_0       --> 0.26.0-np110py27_0

The following packages will be DOWNGRADED due to dependency conflicts:

    numexpr:      2.6.1-np111py27_1       --> 2.6.0-np110py27_0 
    numpy:        1.11.2-py27_0           --> 1.10.4-py27_2     
    scikit-learn: 0.18.1-np111py27_0      --> 0.17.1-np110py27_2
    scipy:        0.18.1-np111py27_0      --> 0.17.1-np110py27_1

Proceed ([y]/n)? n

Exiting

iMac2:tc mrh$ 
```
What am I doing wrong?  It still wants to **downgrade** my version of numpy.

**Comment by PeterDSteinberg** [comment]
Hi @martinholmer @MattHJensen  - I apologize for the package confusion over last couple days.  The issue now is not related to the taxcalc repo (the tags and code do not need to change), but I just need to change the build script that pushes the packages.  It had been hard-coded to look for numpy 1.10 files, then I had built 0.7.9 for both 1.10 and 1.11 and it uploaded wrong one.  Sorry again, I'll update the anaconda packages shortly.

**Comment by martinholmer** [comment]
@PeterDSteinberg said:
> I had built 0.7.9 for both 1.10 and 1.11

Why would you build two versions?
Isn't numpy 1.11.2 the **ONLY** correct numpy version?

**Comment by PeterDSteinberg** [comment]
It had been wrong before (numpy 1.10) and I made a build with 1.10 into my conda-bld directory.  Subsequently we changed the recipe to 1.11, then I built it and had 1.10 and 1.11 packages in by conda-bld directory (old builds are not deleted by default).  The upload script was hardcoded to uploading only the 1.10 version.  Now I have fixed the upload script.

**Comment by PeterDSteinberg** [comment]
Fixed [here on anaconda.org](https://anaconda.org/ospc/taxcalc/files) - Note the package names have `np111` in them.

**Comment by martinholmer** [comment]
@PeterDSteinberg said:
> Fixed here on anaconda.org - Note the package names have np111 in them.

Thanks for fixing this.  It all works as advertised now.

But why all the np111 business in the package name.  That seems redundant.
We're not adding into the taxcalc package name the version of pandas and all the other packages that are being used?  Why even mention the numpy version in the package name?  There is only one correct numpy version, right?

Stuff like this is what may create a feeling among novice taxcalc users that they are in over their heads.

Our goal should be to make it simple for them.  So, only the taxcalc version number is needed, right?

@MattHJensen

**Comment by PeterDSteinberg** [comment]
@martinholmer The file naming is the default style chosen by conda-bld.  It encodes the Python and numpy of the build.

**Comment by martinholmer** [comment]
@PeterDSteinberg said:
> The file naming is the default style chosen by conda-bld. It encodes the Python and numpy of the build.

What's so important about the numpy version?  Why not the pandas version?
In any case, can't you change the default to be more sensible and less confusing to novices?

And where is this default naming style? 
Are you talking about the `tax-calculator/distribution/version_and_release.sh` script?
Why does it still say this:
```
echo "FINISHED CONDA BUILD"
cd $TAXCALC_PACKAGE_DIR
conda convert -p all ~/miniconda3/conda-bld/osx-64/taxcalc-$1-np110py27_0.tar.bz2 -o .
echo "FINISHED CONDA CONVERT"                                 ^^^^^
```
Isn't the `np110` business another bug?

@MattHJensen

**Comment by PeterDSteinberg** [comment]
@martinholmer - Yes that file still has an error - we have a TODO to redo the packaging and have [created this repo policybrain-builder](https://github.com/open-source-economics/policybrain-builder) .  I'll move the logic to that repo this week.

**Comment by martinholmer** [comment]
@PeterDSteinberg said:
> Yes that file still has an error, but I have a different local copy - we have a TODO to redo the packaging and have created this repo `policybrain-builder`.  I'll move the logic to that repo this week.

OK.  That sounds good.

I know removing the `np1xx` (any any other extraneous info) from the package name sounds like a senseless request.  But we are giving serious thought to using the `conda install -c ospc taxcalc` as part of the installation process for first-time users to get started with Tax-Calculator on their local computer.  These people will know nothing about Python or numpy or any of that business.  So, it will be vital to give them an installation process that is as simple as possible.

@MattHJensen

**Comment by PeterDSteinberg** [comment]
@MattHJensen @martinholmer - I have made some progress on a PR to [policybrain-builder here](https://github.com/open-source-economics/policybrain-builder/pull/1) that will help us standardize the conda packaging and anaconda upload.  It has a TODO list. I added a TODO item about looking into simplifying package file names: I agree the long name is potentially overwhelming to new users / non-devs, but it has provided value to us here in realizing we had the wrong numpy version and I'm not sure whether anaconda.org or conda installs have expectations of `np111` being in the file name.
